src/GraphQL/DataLoader/IDataLoaderResult.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace GraphQL.DataLoader
+{
+    /// <summary>
+    /// Represents a pending operation that can return a value
+    /// </summary>
+    /// <typeparam name="T">The type of value that is returned</typeparam>
+    public interface IDataLoaderResult<T> : IDataLoaderResult
```

Comment:
You could have `IDataLoaderResult` extend `IDataLoaderResult<T>` with `IDataLoaderResult<object>` then u can avoid having to create a new method.

Reply:
Then `IDataLoaderResult<T>` cannot be cast to `IDataLoaderResult`.  This causes a major problem as the execution strategy needs to access `IDataLoaderResult`.

This idea was in use with the `ResolveFieldContext` class in the past, which caused a lot of similar problems.  New inheritance patterns going forward are designed with pattern that I've used above.

I also considered making `IDataLoaderResult<T>` covariant.  However, `GetResultAsync()` returns `Task<T>`, and since `Task<T>` is not covariant, `IDataLoaderResult<T>` cannot be covariant.  This is unfortunate, but unavoidable.

I also thought about making `GetResultAsync()` return a `ValueTask<T>`.  This would eliminate allocations for the majority of times that the method is called.  (All but the first time when the data is actually loaded.)  However, the project targets .Net Standard 2.0, where `ValueTask` is not available without introducing a nuget dependency.  The project leads do not wish to add dependencies to the project, so I have left it as a `Task<T>`.  In all fairness, the data loader's fetch function execution time should far outweigh the tiny `Task<T>` allocations necessary to return the data.